src/libraries/System.Private.CoreLib/src/System/Buffers/SharedArrayPool.cs

```diff
@@ -513,4 +521,60 @@ public ThreadLocalArray(T[] array)
             }
         }
     }
+
+    internal static class SharedArrayPoolStatics
+    {
+        /// <summary>Gets the maximum number of partitions to shard arrays into.</summary>
+        /// <remarks>Defaults to int.MaxValue.  Whatever value is returned will end up being clamped to <see cref="Environment.ProcessorCount"/>.</remarks>
+        internal static int GetPartitionCount()
+        {
+            string? value = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_BUFFERS_SHAREDARRAYPOOL_MAXPARTITIONCOUNT");
+            int partitionCount = value is not null && TryParseInt32(value, out int result) && result > 0 ?
+                result :
+                int.MaxValue; // no limit other than processor count
+            return Math.Min(partitionCount, Environment.ProcessorCount);
+        }
+
+        /// <summary>Gets the maximum number of arrays of a given size allowed to be cached per partition.</summary>
+        /// <returns>Defaults to 8. This does not factor in or impact the number of arrays cached per thread in TLS (currently only 1).</returns>
+        internal static int GetMaxArraysPerPartition()
+        {
+            string? value = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_BUFFERS_SHAREDARRAYPOOL_MAXARRAYSPERPARTITION");
```

Comment:
What happens if you set `DOTNET_SYSTEM_BUFFERS_SHAREDARRAYPOOL_MAXARRAYSPERPARTITION` to a very long string and we call into arraypool recursively in `Environment.GetEnvironmentVariable` ?